### ISSUE TITLE
vmalert: replay exit with status 0 when generated samples are not successfully pushed

### ISSUE THREAD

**Issue by Haleygo** [VictoriaMetrics/VictoriaMetrics#6512]
### Describe the bug

Run vmalert replay with unreachable or slow datasource as `remoteWrite.url`, then some generated samples are dropped, but process exits with status 0.

### To Reproduce

```
...
0 / 1 [________________________________________________________________________________] 0.00% ? p/s1 / 1 [-----------------------------------------------------------------------------] 100.00% 10 p/s2024-06-19T16:40:54.268Z	warn	app/vmalert/remotewrite/client.go:251	attempt 1 to send request failed: error while sending request to http://127.0.0.1:18422/api/v1/write: Post "http://127.0.0.1:18422/api/v1/write": dial tcp4 127.0.0.1:18422: connect: connection refused; Data len 1219(1219) (retriable: true)
2024-06-19T16:40:54.838Z	info	app/vmalert/replay.go:71	replay finished! Imported 162 samples
2024-06-19T16:40:55.269Z	warn	app/vmalert/remotewrite/client.go:251	attempt 2 to send request failed: error while sending request to http://127.0.0.1:18422/api/v1/write: Post "http://127.0.0.1:18422/api/v1/write": dial tcp4 127.0.0.1:18422: connect: connection refused; Data len 1219(1219) (retriable: true)
2024-06-19T16:40:57.270Z	warn	app/vmalert/remotewrite/client.go:251	attempt 3 to send request failed: error while sending request to http://127.0.0.1:18422/api/v1/write: Post "http://127.0.0.1:18422/api/v1/write": dial tcp4 127.0.0.1:18422: connect: connection refused; Data len 1219(1219) (retriable: true)
2024-06-19T16:41:01.273Z	warn	app/vmalert/remotewrite/client.go:251	attempt 4 to send request failed: error while sending request to http://127.0.0.1:18422/api/v1/write: Post "http://127.0.0.1:18422/api/v1/write": dial tcp4 127.0.0.1:18422: connect: connection refused; Data len 1219(1219) (retriable: true)
2024-06-19T16:41:09.281Z	warn	app/vmalert/remotewrite/client.go:251	attempt 5 to send request failed: error while sending request to http://127.0.0.1:18422/api/v1/write: Post "http://127.0.0.1:18422/api/v1/write": dial tcp4 127.0.0.1:18422: connect: connection refused; Data len 1219(1219) (retriable: true)
2024-06-19T16:41:24.267Z	warn	app/vmalert/remotewrite/client.go:251	attempt 6 to send request failed: error while sending request to http://127.0.0.1:18422/api/v1/write: Post "http://127.0.0.1:18422/api/v1/write": dial tcp4 127.0.0.1:18422: connect: connection refused; Data len 1219(1219) (retriable: true)
2024-06-19T16:41:24.267Z	error	app/vmalert/remotewrite/client.go:287	attempts to send remote-write request failed - dropping 18 time series
2024-06-19T16:41:24.267Z	info	app/vmalert/remotewrite/client.go:156	shutting down remote write client and flushing remained series
2024-06-19T16:41:24.267Z	info	app/vmalert/remotewrite/client.go:170	shutting down remote write client flushed 0 series
Process 11626 has exited with status 0
```

### Version

all version

### Additional information

I expect the process to exit with error, so user can notice the error immediately and re-run replay(data [deduplication](https://docs.victoriametrics.com/cluster-victoriametrics/#deduplication) should be enabled in victoriametrics).

**Comment by hagen1778** [comment]
https://github.com/VictoriaMetrics/VictoriaMetrics/pull/6513 has been merged and will be available in the next release.

**Comment by valyala** [comment]
This issue must be fixed in [vmalert v1.102.0-rc2](https://github.com/VictoriaMetrics/VictoriaMetrics/releases/tag/v1.102.0-rc2).
